src/main/scala/is/hail/expr/Type.scala

```diff
@@ -465,12 +465,14 @@ object TNumeric {
     if (types.size == 1)
       types.head
     else if (types(TFloat64Required) || types(TFloat64Optional))
-      TFloat64(types(TFloat64Required))
+      TFloat64()
```

Comment:
isn't this wrong? If you have a TFloat64Required and a TInt32Required, it should return TFloat64Required.

Reply:
Agreed with Tim.

Reply:
What I did: nothing in the expr language is designed to handle missingness so I just made it always return optional which matches the previous behavior.  This should get updated when we add missingness throughout expr.

Second, I think you're right: it should be optional if argument is optional, otherwise, required, and the type should be determined the old way, but I wasn't 100% and wanted to double-check how this is used.  But again, it doesn't matter until expr is missing-aware.